lib/string-checker.js

```diff
@@ -126,6 +133,41 @@ StringChecker.prototype = {
     },
 
     /**
+     * Fix provided error.
+     *
+     * @param {JsFile} file
+     * @param {Errors} errors
+     * @protected
+     */
+    _fixJsFile: function(file, errors) {
+        var list = errors.getErrorList();
+        var configration = this.getConfiguration();
+
+        list.forEach(function(error) {
+            if (error.fixed) {
+                return;
+            }
+
+            var instance = configration.getConfiguredRule(error.rule);
+
+            if (instance && instance.fix) {
+                try {
+
+                    // "error.fixed = true" should go first, so rule can
+                    // decide for itself (with "error.fixed = false")
+                    // if it can fix this particular error
+                    error.fixed = true;
```

Comment:
> I guess an error being unfixable isn't an exception?

Yeah, otherwise we would show that big message, inviting user to submit a bug to this repo.

> What I'm getting at is avoiding rules knowing about the fixed property of the error.

Alternative for it would be to make a return value of the `fix` method count, if returns `undefined | true` -> `error.fixed = true` if returns `false` -> `error.fixed = undefined`, or `false` value, which, btw could signify failure (either through exception or explicit return value) of fix attempt.